wiki/Concept-Starting-and-orchestrating-Bots.md

```diff
@@ -22,17 +25,65 @@ I would recommend an infrastructure that would look like the following diagram:
 
 Con:
-* Would require additional authentification issues related to running the Bots on the local machine of different users, such as "Who should be allowed to run a Bot on whose computer?"
-* In my opinion this only makes sense **after** the foundation has been laid through the possibility of a single person running multiple Bots at the same time, which as far as my understanding and knowledge goes, would be when the Cloud is available for running the Bots in there
-* Outsourcing the orchestration would allow focusing more on the core functionality and should be fairly easy to do once the APIs have been opened to external services to start Bots
+
+- Would require additional authentification issues related to running the Bots on the local machine of different users, such as "Who should be allowed to run a Bot on whose computer?"
+- In my opinion this only makes sense **after** the foundation has been laid through the possibility of a single person running multiple Bots at the same time, which as far as my understanding and knowledge goes, would be when the Cloud is available for running the Bots in there
+- Outsourcing the orchestration would allow focusing more on the core functionality and should be fairly easy to do once the APIs have been opened to external services to start Bots
 
 My recommendation would be to not take care of the orchestration in-house, but rather let external services take care of that (for now). I can see the benefits, which would especially help the management level through statistics and extended information, but I do not think that the orchestration should be of priority because of the rabbit hole that we might fall into security-wise.
+
+# Communication with ark automate dektop app
+
+I looked at several possible technologies/concepts that be applied in our case.
+
+1. Polling and Long-Polling
+2. Websockets
+3. Server-sent events
+4. Higher Abstraction Frameworks
+
+### Polling and Long-Polling
+
+Polling is an approach where the desktop app would permenantly 'ask' the there is a new robot to execute. Although it is quite easy to implement, the performance is bad.
+Long Polling aims to solve the performance issue by just 'asking' one time and waiting for an answer. The desktop app would 'ask' the server again after a fixed periode of time if the server does not have any new robots to execute. In principle a great approach but it lacks support of frameworks one could use and implementing it from scretch would be too time consuming
+
+### Websockets
+
+When using websockets we would open a bidirectional communication channel between the server side and the client side.Thus a great solution because real time communication is possible. Still, it is time consuming to implement and a successful connection is not always garantued as either element (proxy, firewall, …) prevent WebSocket connections between the client and the server or the browser does not support it yet.
```

Comment:
```suggestion
When using websockets we would open a bidirectional communication channel between the server side and the client side. This is a great solution because real time communication is possible. Still, it is time consuming to implement and a successful connection is not always guaranteed as either element (proxy, firewall, …) prevent WebSocket connections between the client and the server or the browser does not support it yet.
```